assets/js/util.js

```diff
@@ -293,15 +298,74 @@ function callApy(options, endpoint) {
     var requestUrl = window.location.protocol + window.location.hostname +
         window.location.pathname + '?' + $.param(requestOptions.data);
 
+    callApyStartTime = Date.now();
     if(requestUrl.length > THRESHOLD_REQUEST_URL_LENGTH) {
         requestOptions.type = 'POST';
         return $.ajax(requestOptions);
     }
     return $.jsonp(requestOptions);
 }
 
+function callApyDuration() {
+    callApyEndTime = Date.now();
+    checkServiceLoadTimes(callApyEndTime - callApyStartTime);
+}
+
+function checkServiceLoadTimes(requestDuration) {
+    var individualDurationThreshold = 2000;
+    var cumulativeDurationThreshold = 1500;
+    var demoThreshold = 10000; // for demo purposes only
+    if(store.able()) {
+        if(sessionStorage.requestsMade) {
+            sessionStorage.requestsMade = Number(sessionStorage.requestsMade) + 1;
+        }
+        else {
+            sessionStorage.requestsMade = 1;
+        }
+
+        if(sessionStorage.cumulativeRequestsTime) {
+            sessionStorage.cumulativeRequestsTime = Number(sessionStorage.cumulativeRequestsTime) + requestDuration;
+        }
+        else {
+            sessionStorage.cumulativeRequestsTime = requestDuration;
+        }
+
+        var averageRequestsDuration = (sessionStorage.cumulativeRequestsTime) / (sessionStorage.requestsMade);
+
+        if(requestDuration > individualDurationThreshold || averageRequestsDuration > cumulativeDurationThreshold) {
```

Comment:
Um, oh! So if there is a threshold gap (say 5 min)  between 2 requests, I will start all of this afresh. Makes sense? This will ensure that our time window is moving.